railties/lib/rails/commands/server/server_command.rb

```diff
@@ -165,6 +165,14 @@ def restart_command
           "bin/rails server #{@server} #{@original_options.join(" ")}"
         end
 
+        def pid
+          if options[:pid] == DEFAULT_PID_PATH
+            File.expand_path(DEFAULT_PID_PATH)
+          else
+            options[:pid]
```

Comment:
I did this according to the original code. But you are right. 
If it does not expand it may cause error in daemon mode. I fixed.